apps/studio-next/src/middleware.ts

```diff
@@ -0,0 +1,30 @@
+import { NextRequest, NextResponse, userAgent } from "next/server";
+import crawlers from 'crawler-user-agents';
+
+export async function middleware(request: NextRequest) {
+  const userAgents = crawlers.map(crawler => crawler.pattern);
+  const requestInfo = userAgent(request);
+  const res = NextResponse.next();
+  const documentURL = request.nextUrl.searchParams.get("url");
+
+  for (const ua of userAgents) {
+    if (requestInfo.ua.toLowerCase().includes(ua.toLowerCase())) {
+      const encodedDocument = request.nextUrl.searchParams.get("base64");
+      
+      if (!encodedDocument && !documentURL) {
+        return res;
+      }
+      if (encodedDocument) {
+        return NextResponse.rewrite(new URL(`/api/crawler?base64=${encodedDocument}`, request.url));
+      }
+      if (documentURL) {
+        return NextResponse.rewrite(new URL(`/api/crawler?url=${documentURL}`, request.url));
+      }
+    }
+  }
+  return res;
+}
+
+export const config = {
+  matcher: ['/:base64', '/:url'],
+}
```

Comment:
Is this matcher configuration correct? I feel like it is matching to path parameters instead of query parameters that we use for url and base64.

Reply:
Oh yes, you are right, I didn't interpret the [documentation](https://nextjs.org/docs/app/building-your-application/routing/middleware#matching-paths) correctly, I will make the changes.